xbmc/platform/android/network/NetworkAndroid.h

```diff
@@ -72,4 +69,10 @@ class CNetworkAndroid : public CNetworkBase
   std::vector<CNetworkInterface*> m_interfaces;
   std::vector<CNetworkInterface*> m_oldInterfaces;
   CCriticalSection m_refreshMutex;
+
+  CNetworkInterface* m_defaultInterface = nullptr;
```

Comment:
`nullptr` is implicit here no need for the assignment

Reply:
Ok, I remove it

Reply:
Is it? It should be an arbitrary value. So from my pov it's needed for proper null initialization. Or did something change in recent years I am not aware of? :-)

Reply:
I think it's only not needed when using a smart pointer.

Reply:
```
fritsch@l380:~$ ./Test4.out 
0x7f837a11e754
fritsch@l380:~$ ./Test4.out 
0x7fb4f791e754
fritsch@l380:~$ ./Test4.out 
0x7fd52f51e754
```
via:

```
#include <iostream>

int main()
{
  int* dangle;
  std::cout << dangle << std::endl;
  return 0;
}
```

But maybe I just did get up too early?

Reply:
And I agree with @lrusak - normal ptrs always have to be initialized else they dangle.

Reply:
You're right, I add it again, thanks

Reply:
> I think it's only not needed when using a smart pointer.

Yeap, brain fart from my side. Sorry.